app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt

```diff
@@ -531,8 +531,9 @@ abstract class BaseBrowserFragment : Fragment(), BackHandler, SessionManager.Obs
             } else {
                 val isLastSession =
                     sessionManager.sessionsOfType(private = session.private).count() == 1
-                sessionManager.remove(session, true)
-                return !isLastSession // Jump to tab overview if last session was removed
+                sessionManager.remove(session, session.hasParentSession)
+                val dontGoToOverview = !(isLastSession || !session.hasParentSession)
```

Comment:
Usually it's easier to read booleans when they represent the "true" state.

```kotlin
val goToOverview = isLastSession || !session.hasParentSession
return !goToOverview
```

Reply:
Good point, updated.